drivers/SmartThings/philips-hue/src/handlers.lua

```diff
@@ -265,28 +267,92 @@ local function do_refresh_light(driver, light_device)
   local success = false
   local count = 0
   local num_attempts = 3
+  local rest_resp, rest_err = nil, nil
+  local zigbee_resource_id = nil
+  --- this loop is a rate-limit dodge.
+  ---
+  --- One of the various symptoms of hitting the Hue Bridge's rate limit is that you'll get a silent
+  --- failure that takes the form of the bridge returning the last valid response it replied with.
+  --- So we hit the bridge 2-3 times and check the IDs in the responses to verify that we're getting
+  --- the information for the light that we expect to getting the info for.
```

Comment:
would it make sense to do a backoff/sleep/whatever if we notice that this has happened?

Reply:
Probably but I didn't want to make changes outside of the scope of what was needed to unblock the firmware release.